code/modules/library/lib_machines.dm

```diff
@@ -213,3 +214,109 @@ var/global/list/library_section_names = list("Fiction", "Non-Fiction", "Adult",
 			QDEL_NULL(O)
 	else
 		return ..()
+
+var/global/datum/research/research_archive_datum
+
+/obj/machinery/researcharchive
+	name = "research archive"
+	desc = "A high-powered data archive device that takes technology disks and persistently backs them up to specialized servers for the upcoming shift. Usually takes two disks per technology."
+	icon = 'icons/obj/library.dmi'
+	icon_state = "binder"
+	anchored = TRUE
+	density = TRUE
+	machine_flags =  WRENCHMOVE | FIXED2WORK | EJECTNOTDEL // | SCREWTOGGLE | CROWDESTROY
+	pass_flags = PASSTABLE
+	idle_power_usage = 4
+	var/obj/item/weapon/disk/tech_disk/diskslot
+	var/busy = FALSE
+
+/obj/machinery/researcharchive/New()
+	..()
+	if(!research_archive_datum)
+		research_archive_datum = new /datum/research()
+
+/obj/machinery/researcharchive/examine(mob/user)
+	..()
+	if(diskslot)
+		to_chat(user,"<span class='info'>In the slot you can see a disk that contains [diskslot.stored.id] [diskslot.stored.level].</span>")
+		diskslot.examine(user)
+
+/obj/machinery/researcharchive/attackby(var/obj/item/weapon/W, var/mob/user)
+	if(stat & (BROKEN))
+		to_chat(user, "<span class='warning'>\The [src] is broken!</span>")
+		return
+	if(..())
+		return
+
+	if(busy)
+		return
+
+	if (!istype(W,/obj/item/weapon/disk/tech_disk))
+		to_chat(user, "<span class='warning'>\The [src] only accepts technology disks.</span>")
+		return
+
+	if (!user.drop_item(W, src))
+		return
+	if(diskslot)
+		user.put_in_hands(diskslot)
+		visible_message("<span class='notice'>\The [user] swaps the disks in \the [src].</span>","<span class='notice'>You swap the disks in \the [src].</span>")
+	else
+		visible_message("<span class='notice'>\The [user] adds \the [W] to \the [src].</span>","<span class='notice'>You add \the [W] to \the [src].</span>")
+	diskslot = W
+	playsound(loc, 'sound/machines/click.ogg', 50, 1)
+	update_icon()
+
+/obj/machinery/researcharchive/attack_hand(var/mob/user)
+	. = ..()
+	if(stat & (BROKEN))
+		to_chat(user, "<span class='notice'>\The [src] is broken.</span>")
+		return
+
+	if(stat & (NOPOWER))
+		to_chat(user, "<span class='notice'>\The [src] is unpowered.</span>")
+		return
+
+	if(!diskslot)
+		to_chat(user, "<span class='notice'>There is no inserted technology disk.</span>")
+		return
+
+	if(busy)
+		return
+
+	if(diskslot.stored.id in list("syndicate", "Nanotrasen", "anomaly"))
+		to_chat(user, "<span class='notice'>\The [src] cannot process this technology data due to proprietary encoding.</span>")
+		return
+
+	playsound(loc, "sound/machines/heps.ogg", 50, 1)
+
+	busy = TRUE
+	use_power(200)
+	spawn(3 SECONDS)
```

Comment:
Empty `spawn()`
Was this meant to be `sleep()` or is there a missing block here?